app/javascript/responseTemplates/responseTemplates.js

```diff
@@ -0,0 +1,249 @@
+      'X-CSRF-Token': window.csrfToken,
+      'Content-Type': 'application/json',
+    },
+  })
+    .then((response) => response.json())
+    .then((response) => {
+      form.querySelector('img.loading-img').classList.toggle('hidden');
+      dataContainer.innerHTML = buildHTML(response, typeOf);
+      const topLevelData = document.getElementById('response-templates-data')
+      topLevelData.innerHTML = dataContainer.parentElement.innerHTML
+      addClickListeners(form);
+    });
+}
+
+function prepareHeaderButtons(form) {
+  const personalTemplateButton = form.querySelector('.personal-template-button');
+  const modTemplateButton = form.querySelector('.moderator-template-button');
+
+  personalTemplateButton.addEventListener('click', (e) => {
+    toggleTemplateTypeButton(form, e)
+  });
+  modTemplateButton.addEventListener('click', (e) => {
+    toggleTemplateTypeButton(form, e)
+  });
+  modTemplateButton.classList.remove('hidden');
+
+  modTemplateButton.addEventListener('click', () => {
+    const topLevelData = document.getElementById('response-templates-data')
+    const modDataNotFetched = topLevelData.innerHTML !== ''
+      ? topLevelData.querySelector('.moderator-responses-container')
+          .childElementCount === 0
+      : false;
+    if (modDataNotFetched) {
+      form.querySelector('img.loading-img').classList.toggle('hidden')
+      fetchResponseTemplates('mod_comment', form.id)
+    }
+    }, { once: true });
+}
+
+function copyData(responsesContainer) {
+  responsesContainer.innerHTML = document.getElementById(
+    'response-templates-data',
+  ).innerHTML;
+}
+
+function openButtonCallback(form) {
+  const responsesContainer = form.querySelector('.response-templates-container');
+  const dataFetched = document.getElementById('response-templates-data').innerHTML !== ''
+
+  responsesContainer.classList.toggle('hidden');
+
+  if (dataFetched && !responsesContainer.classList.contains('hidden')) {
+    copyData(responsesContainer)
+    addClickListeners(form)
+  } else {
+    form.querySelector('img.loading-img').classList.toggle('hidden');
+    fetchResponseTemplates('personal_comment', form.id);
+  }
+  /* eslint-disable-next-line no-undef */
+  if (userData().moderator_for_tags.length > 0) {
+    prepareHeaderButtons(form);
+  }
+}
+
+export function prepareOpenButton(form) {
+  const button = form.querySelector('.response-templates-button')
+  if (!button) {
+    return
+  }
+
+  button.addEventListener('click', () => {
+    openButtonCallback(form)
+  });
+}
+
+export function observeForReplyClick() {
+  const config = { childList: true, subtree: true };
+
+  const callback = (mutations) => {
+    const form = mutations[0].addedNodes[0];
+    if (form.nodeName === "FORM") {
+      prepareOpenButton(form)
+    }
+  };
+
+  const observer = new MutationObserver(callback);
+
+  const commentTree = document.getElementById('comment-trees-container')
+  observer.observe(commentTree, config)
```

Comment:
@nickytonline observe one thing instead of `n` reply buttons. :D

Referencing: https://github.com/thepracticaldev/dev.to/pull/5419#discussion_r380373267

One thing to consider though is that it constantly observes the tree `div`. Not sure if that's expensive or not, but it won't do anything if the observed change's `addedNode` is not a `form`.

Reply:
The way `MutationObserverver` works is it's constantly monitoring any changes for the type you set in its configuration. That's the only reason I brought it up. 😉